### ISSUE TITLE
Environment.TickCount

### ISSUE THREAD

**Issue by monokot** [fbarresi/Sharp7#15]
Hello!

We are using an implementation for C# and I would like to draw your attention to the following code:

```
int Elapsed = Environment.TickCount;
...
Expired = Environment.TickCount - Elapsed > Timeout;
```

This is because Environment.TickCount returns values from Int32.MinValue to Int32.MaxValue. Therefore, if the system has been running for 24.9 days, the counter will turn negative and the expression Expired = Environment.TickCount-Elapsed > Timeout will return false for the next 24.9 days.
​
Instead of Environment.TickCount, I would suggest using a static instance of Stopwatch (which is run once) to get uptime milliseconds.

**Comment by fbarresi** [comment]
Hi!

I guess, you are referring to [this line](https://github.com/fbarresi/Sharp7/blob/d662df64fde38ede39b3ad585503e084b300d566/Sharp7/MsgSocket.cs#L84), right?

This method would cause a single transmission error if the TickCount overflows exactly while waiting for data and I think any application resilient enough will keep working anyway.
Honestly I use sharp7 in many application that runs in high frequency polling over many years and I didn't get any problem at this point.

I understand the problem, I will fix it.
Thank you for your suggestion.

Best regards, 

Federico

**Comment by Peter-B-** [comment]
I don't think that this will cause an issue due to the way [.net handles integer arithmetic](https://stackoverflow.com/a/1078089/909237).
